proposals/left-right-join-in-query-expressions.md

```diff
@@ -0,0 +1,235 @@
+    from student in students
+    join department in departments on student.DepartmentID equals department.ID into gj
+    from subgroup in gj.DefaultIfEmpty()
+    select new
+    {
+        student.FirstName,
+        student.LastName,
+        Department = subgroup?.Name ?? string.Empty
+    };
+```
+
+Or using method syntax:
+
+```c#
+var query = students
+    .GroupJoin(departments, student => student.DepartmentID, department => department.ID, (student, departmentList) => new { student, subgroup = departmentList })
+    .SelectMany(
+        joinedSet => joinedSet.subgroup.DefaultIfEmpty(),
+        (student, department) => new
+        {
+            student.student.FirstName,
+            student.student.LastName,
+            Department = department.Name
+        });
+```
+
+Although functionality sufficient for expressing a left join operation, this combining approach has the following drawbacks:
+
+- It's complicated, requiring combining multiple different LINQ operators in a specific way to form a complex construct, and is easy to accidentally get wrong. Many EF users have complained about the complexity of this construct for expressing a simple SQL LEFT JOIN.
+- It is inefficient - the combination of operators adds significant overhead compared to a single operator using an internal lookup table (or "hash join", as Join is implemented).
+
+https://github.com/dotnet/runtime/issues/110292 is an approved API proposal for introducing LeftJoin and RightJoin operators into the System.Linq for .NET 10; see that proposal for additional performance information and benchmarks. Aside from allowing users to more easily express SQL LEFT/RIGHT JOIN when using a LINQ provider (such as EF Core), it would also allow both easier and more efficient in-memory left/right joins, exactly in the way that inner joins are already supported.
+
+Along with the introduction of the operators into System.Linq, the C# query expression syntax can be extended with new `left join` and `right join` clauses which would translate to these new operators, allowing for simpler and more efficient code where C# query syntax is used.
+
+## Detailed design
+[design]: #detailed-design
+
+Proposed grammar change ([§11.7.1](https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/expressions.md#1117-query-expressions)):
+
+### Grammar
+
+```diff
+join_clause
+-    : 'join' type? identifier 'in' expression 'on' expression
++    : ('left' | 'right')? 'join' type? identifier 'in' expression 'on' expression
+      'equals' expression
+    ;
+```
+
+As per the grammar proposal, the syntax for the proposed `left join` and `right join` expression clauses would be identical to the existing `join` clause. For example:
+
+```c#
+from student in students
+left join department in departments on student.DepartmentID equals department.ID
+select new { student.Name, department?.Name }
+```
+
+### Left join translation
+
+A `left join` clause immediately followed by a `select` clause
```

Comment:
Mention which section of the spec (by number) (and with a link) that you are updating.  https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/expressions.md#111735-from-let-where-join-and-orderby-clauses